### DISCUSSION TITLE
Ignore files with upload()

### DISCUSSION THREAD

**Discussion by patrickhoehn** [deployphp/deployer#2340] [Help needed]
Is there a way to ignore files from being uploaded with `upload()`? Right now I'm uploading everything and delete the not needed files/folders afterwards

**Comment by antonmedv** [comment]
Yes, basically it rsync, you can pass options to it.

**Reply by patrickhoehn** [reply]
Great! Thank you!

**Comment by patrickhoehn** [comment]
@antonmedv, is this the right usage? 
`task('upload', function () {
    upload(__DIR__ . "/", '{{release_path}}', ['options' => '--exclude "node_modules"']);
});`
because it doesn't work and don't show up in verbose output: 
`rsync -azP -e 'ssh -A -i ~/.ssh/id_rsa' ...`

**Reply by patrickhoehn** [reply]
Ok, putting brackets around `'--exclude "node_modules"'` fixed it!

**Reply by antonmedv** [reply]
Maybe add support for options as string? Can you send pr?

**Reply by kdekooter** [reply]
@patrickhoehn would you be so kind as to post the final correct syntax? Totally confused by your "brackets" remark 🤯.

This is my - not working - code:
```
task('upload', function () {
  upload(__DIR__ . "/", '{{release_path}}', [ 'options' => '--exclude \'frontend\'']);
});
```

**Reply by patrickhoehn** [reply]
@kdekooter, my last working stage on this project was:
`task('upload', function () {
    upload(__DIR__ . "/", '{{release_path}}', ['options' => ['--exclude-from="deploy-exclude.txt"']]);
});` so I guess I meant those brackets (bold) ['options' => **[**'--exclude-from="deploy-exclude.txt"'**]**]. I hope this is still useful for you
